gluten-core/src/main/java/io/glutenproject/memory/memtarget/spark/TreeMemoryConsumers.java

```diff
@@ -62,7 +64,8 @@ public static Factory shared() {
 
   public static class Factory {
 
-    private static final WeakHashMap<TaskMemoryManager, TreeMemoryTarget> MAP = new WeakHashMap<>();
+    private static final ReferenceMap<TaskMemoryManager, TreeMemoryTarget> MAP =
+            new ReferenceMap<>(ReferenceStrength.WEAK, ReferenceStrength.WEAK);
```

Comment:
This looks very suitable for our case except I have one single question: Since the map key is `tmm` and map value is usually `TreeMemoryConsumer(tmm)`, would it mean the map value could ideally get garbage-collected before the corresponding key? When this happens, is it possible to have `null` returned from `#getSharedAccount` which lead to NPE?

Reply:
As per the documentation of org.apache.commons.collections4.map.ReferenceMap: [If non-hard references are used, then the garbage collector can remove mappings if a key or value becomes unreachable](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/ReferenceMap.html#:~:text=If%20non%2Dhard%20references%20are%20used%2C%20then%20the%20garbage%20collector%20can%20remove%20mappings%20if%20a%20key%20or%20value%20becomes%20unreachable)

If value is garbage collected, both key value pair will be removed.